networkx/algorithms/assortativity/tests/test_mixing.py

```diff
@@ -71,9 +68,7 @@ def test_degree_mixing_matrix_multigraph(self):
 
     def test_degree_mixing_matrix_selfloop(self):
         # fmt: off
-        a_result = np.array([[0, 0, 0],
-                             [0, 0, 0],
-                             [0, 0, 2]]
+        a_result = np.array([[2]]
```

Comment:
The same reasoning. Here is a self loop graphs. It consists of 2 nodes with 2 self loops. Both nodes have degree 2. Then, the number of unique degrees is 1 and the mixing matrix should be 1x1.